src/render/OpenGL.cpp

```diff
@@ -1755,8 +1757,19 @@ void CHyprOpenGLImpl::createBGTextureForMonitor(CMonitor* pMonitor) {
     m_mMonitorRenderResources[pMonitor].backgroundTexBox = box;
 
     // create a new one with cairo
-    const auto CAIROSURFACE = cairo_image_surface_create_from_png(texPath.c_str());
-    const auto CAIRO        = cairo_create(CAIROSURFACE);
+    cairo_surface_t* CAIROSURFACE;
+    if (texPath.empty())
+        // if we couldn't find a texture, use a grey filler image instead.
+        CAIROSURFACE = cairo_image_surface_create(CAIRO_FORMAT_ARGB32, textureSize.x, textureSize.y);
+    else
+        CAIROSURFACE = cairo_image_surface_create_from_png(texPath.c_str());
+    const auto CAIRO = cairo_create(CAIROSURFACE);
+
+    if (texPath.empty()) {
+        cairo_set_source_rgba(CAIRO, 0.106, 0.106, 0.106, 1.0);
```

Comment:
there is a setting for a default bg col. We should leave this empty anyways and not render it if textpath is missing to save on a draw call.

Reply:
The image is cached so the draw call is kind of a moot point. As for the bg color, have you considered making the logo images transparent so they can be overlayed? I just took the background color from the images.

Reply:
rendering of the texture is another heavy drawcall especially on lower-end hardware. We also disable blending to save on some compute time.

Currently, last time I was profiling, rendering the background texture was one of the heaviest calls in the entire draw stack.

Reply:
That sounds somewhat concerning, but I think fixing that should be a seperate change based on occlusion of the background (e.g. hyprpaper), and not randomly disabling the background image if we can't find it, because that means the previous frame is never cleared.

Reply:
if the image is not found we should release the tex and just check for that here:
https://github.com/hyprwm/Hyprland/blob/19c4855afce7e4dfa316cbd40f2c991fd3f87e0b/src/render/Renderer.cpp#L1021-L1024

Reply:
Unless that's new since this pr, it dosen't actually work. Also that bit looks like it wouldn't use the background image if the texture exists which seems wrong. You also lose the splash, but I see your point.

What do you think of always doing the bkg color clear and using transparent bkg images?
Also not sure why the bkg clear would be so expensive unless it triggers unconditionally every frame.

Reply:
IMO if you want to make not clearing the background an option, it should be a clearly defined configuration option and not removing an undocumented system file. In my opinion attempting to match the behavior with the image if its missing is the best solution.

Reply:
Oh sorry I was confusing it with the old behavior. I'll look at it again tomorrow. Using clear seems fine but imo the background color with an image should match for consistency.